monkey/common/credentials/nt_hash.py

```diff
@@ -1,23 +1,16 @@
-from dataclasses import dataclass, field
+import re
 
-from marshmallow import fields
+from pydantic import validator
 
-from . import CredentialComponentType, ICredentialComponent
-from .credential_component_schema import CredentialComponentSchema, CredentialTypeField
-from .validators import credential_component_validator, ntlm_hash_validator
+from ..base_models import InfectionMonkeyBaseModel
+from .validators import ntlm_hash_regex
 
 
-class NTHashSchema(CredentialComponentSchema):
-    credential_type = CredentialTypeField(CredentialComponentType.NT_HASH)
-    nt_hash = fields.Str(validate=ntlm_hash_validator)
-
-
-@dataclass(frozen=True)
-class NTHash(ICredentialComponent):
-    credential_type: CredentialComponentType = field(
-        default=CredentialComponentType.NT_HASH, init=False
-    )
+class NTHash(InfectionMonkeyBaseModel):
     nt_hash: str
 
-    def __post_init__(self):
-        credential_component_validator(NTHashSchema(), self)
+    @validator("nt_hash")
+    def validate_hash_format(cls, nt_hash):
+        if not re.match(ntlm_hash_regex, nt_hash):
+            raise ValueError(f"Invalid nt hash provided: {nt_hash}")
```

Comment:
This is not duplication, because it's validating different things. It just happens that we use the same regex for both. I looked online and couldn't find a definitive answer for the format of lm hash, for example [here](https://medium.com/@petergombos/lm-ntlm-net-ntlmv2-oh-my-a9b235c58ed4) the lm hash example is a lot shorter. I don't think it matters that much, the LM hash is very old and I'm not even sure we could propagate with it. With that in mind, once we decide to remove it there will be a validation method on a separate file for no reason

Reply:
> once we decide to remove it there will be a validation method on a separate file for no reason

Once we decide to remove it, we can change that as well.

Reply:
It will be harder to change, because we'll need to move it back. So why even bother? What do we gain?